### DISCUSSION TITLE
Use as dependency to generate automatically build from a Spring Project

### DISCUSSION THREAD

**Discussion by CalderonRigoberto** [schemaspy/schemaspy#1423] [Q&A]
Hello friends, 

I want to put on a personal site the output generated with this amazing schemaspy tool, but I have doubts how to do it in a simple way. I already have it with a dockerfile in Jenkins using Envinject and also Parametrized Trigger plugin. But this requires me more configuration work due to the fact that I want to do it in multiple projects.

I would like to create a utility class that I don't know if it already exists to run the schemaspy arguments in a java project taking the arguments from the .properties file, so that I only have to take the generated output and deploy it to the site where I will host the results for collaborators to see.

I was planning something like code below. 
```java
Properties properties = new Properties();
try (FileInputStream fis = new FileInputStream("path/to/schemaSpy.properties")) {
  properties.load(fis);
}
List<String> command = new ArrayList<>();
command.add("java");
command.add("-jar");
command.add("path/to/schemaspy.jar");

for (String key : properties.stringPropertyNames()) {
  command.add("-" + key);
  command.add(properties.getProperty(key));
}

command.add("-o");
command.add("target/output-directory");

String[] commandArray = command.toArray(new String[0]);
```

And after use  
```java
 schemaSpyRunner.run(commandArray);
```

I'm searching for advices, I'm nobbie on this, so I dont know what is the best alternative to generate this on multiple projects.

Regards,

**Comment by npetzall** [comment]
We already have the SchemaSpy.properties that is read automatically, else it can be specified with `-configFile` so you can keep everything in any file. So one file per project for instance, by having each project have it's own filename, `-configFile projA.properties`, but that has more to do with organisation and I know nothing of you're layout.

The it's a matter of when and how to run schemaspy, if docker you'll need at least two mounts, 1 for the configFile and one for output.

Depending on how you manage the database and it's setup you could either build with each build or you could build each time the database is changed.

There is also the possibility of running it just based of script like 
https://github.com/tmfg/digitraffic-road/tree/master/dbroad/schemaspy 

I would try to avoid adding SchemaSpy to you classpath, unless you want to be able to generate new reports during runtime. Else I would keep it of the classpath just to avoid conflict with versions or accidentally creating a runtime dependency on SchemaSpy.

**Reply by npetzall** [reply]
There's also this comment I've made

https://github.com/schemaspy/schemaspy/pull/1419#issuecomment-1975254880

SchemaSpy also checks ENV's but it's undocumented and I would say unfinished. That might be a possibility too, that you have a basic SchemaSpy.properties that is missing env/project specific details and that the "overrides" are with ENV's, when ENV's has been finished/cleaned up/corrected/works properly.
